src/registry.ts

```diff
@@ -0,0 +1,41 @@
+import { Document } from "@govtechsg/open-attestation";
+import { ethers, Signer } from "ethers";
+import { getBatchMerkleRoot } from "./util/token";
+import { abi as TokenRegistryABI } from "../build/contracts/ERC721MintableFull.json";
+
+export class TokenRegistry {
+  web3Provider: Signer;
+
+  address: string;
+
+  contractInstance: any;
+
+  /**
+   * Creates a TokenRegistry instance with the specified address and ethersjs signer
+   * @param {object} instanceParameters
```

Comment:
you shouldnt type the parameter (easy to make the type non synchronized)

Reply:
ok